dspace-server-webapp/src/test/java/org/dspace/app/rest/authorization/EPersonRegistrationFeatureIT.java

```diff
@@ -0,0 +1,111 @@
+
+public class EPersonRegistrationFeatureIT extends AbstractControllerIntegrationTest {
+
+    @Autowired
+    private AuthorizationFeatureService authorizationFeatureService;
+
+    @Autowired
+    private ConverterService converterService;
+
+    @Autowired
+    private ConfigurationService configurationService;
+
+    @Autowired
+    private SiteService siteService;
+
+    @Autowired
+    private SiteConverter siteConverter;
+
+    @Autowired
+    private Utils utils;
+
+    private AuthorizationFeature epersonRegistrationFeature;
+
+    public static final String[] SHIB_ONLY = {"org.dspace.authenticate.ShibAuthentication"};
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        epersonRegistrationFeature = authorizationFeatureService.find(EPersonRegistrationFeature.NAME);
+    }
+
+    @Test
+    public void userRegistrationEnabledSuccessTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isOk());
+    }
+
+    @Test
+    public void userRegistrationDisabledUnAuthorizedTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+        configurationService.setProperty("user.registration", false);
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isNoContent());
+
+    }
+
+
+    @Test
+    public void userRegistrationEnabledShibTest() throws Exception {
+
+        Site site = siteService.findSite(context);
+        SiteRest SiteRest = siteConverter.convert(site, Projection.DEFAULT);
+        String siteUri = utils.linkToSingleResource(SiteRest, "self").getHref();
+
+
+        getClient().perform(get("/api/authz/authorizations/search/objectAndFeature")
+                                          .param("uri", siteUri)
+                                          .param("feature", epersonRegistrationFeature.getName()))
+                             .andExpect(status().isOk());
+
+        //Enable Shibboleth and password login
+        configurationService.setProperty("plugin.sequence.org.dspace.authenticate.AuthenticationMethod", SHIB_ONLY);
```

Comment:
I don't think that this is sufficient to get shibboleth enabled as the plugins were already read and so cached

Reply:
If you compare the check on top and below, it does impact the behavior.
https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/authenticate/AuthenticationServiceImpl.java#L70 doesn't seem to be cached either

Reply:
you are right. Looking to the org.dspace.core.LegacyPluginServiceImpl#getPluginSequence it seems that there is a bug or at least the comments don't align with the implementation as there is no cache for the plugins sequence... cache that is indeed here for the other plugin types. I'm going to create a ticket for that but it is not related to this PR, so I'm flagging this thread as solved